### PR TITLE
[ML][Inference] add new flag for optionally including model definition

### PR DESCRIPTION
**Pull request by benwtrent**
This adds a new flag `include_model_definition` that includes the model definition on a GET call. 

By default the definition is not returned.

Additionally, only ONE config is allowed to be returned when this flag is set.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java [hunk 1]
```diff
@@ -216,6 +288,92 @@ public void deleteTrainedModel(String modelId, ActionListener<Boolean> listener)
...
+    private QueryBuilder buildQueryIdExpressionQuery(String[] tokens, String resourceIdField) {
+        BoolQueryBuilder boolQuery = QueryBuilders.boolQuery()
+            .filter(QueryBuilders.termQuery(InferenceIndexConstants.DOC_TYPE.getPreferredName(), TrainedModelConfig.NAME));
+
+        if (Strings.isAllOrWildcard(tokens)) {
+            return boolQuery;
+        }
+        // If the resourceId is not _all or *, we should see if it is a comma delimited string with wild-cards
+        // e.g. id1,id2*,id3
+        BoolQueryBuilder shouldQueries = new BoolQueryBuilder();
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java]: We are expanding the ids in other parts of our codebase. Shouldn't this logic (splitting by comma, wildcards) be reused?

Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java]: @przemekwitek I agree we should have some id expansion class, but the individual pieces are slightly different enough that finding the appropriate abstraction would take some refactoring. I would rather not do that type of refactoring inside of a feature branch..

**Review by przemekwitek — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/bwc
run elasticsearch-ci/default-distro

**Pull request merged**